sdk/tables/azure-data-tables/migration_guide.md

```diff
@@ -0,0 +1,309 @@
+# Guide for migrating to `azure-data-tables` from `azure-cosmosdb-table`
+
+This guide is intended to assist in the migration to `azure-data-tables` from `azure-cosmosdb-table`. It will focus on side-by-side comparisons for similar operations between the two packages.
+
+We assume that you are familiar with `azure-cosmosdb-table`. If not, please refer to the README for [`azure-data-tables`](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/tables/azure-data-tables#azure-data-tables-client-library-for-python) rather than this guide.
+
+## Table of contents
+
+* [Migration benefits](#migration-benefits)
+* [Important changes](#important-changes)
+    - [Package Names](#package-names-and-namespaces)
+    - [Client hierarchy and constructors](#client-hierarchy-and-constructors)
+    - [Authenticating Clients](#authenticating-clients)
+    - [Table Level Scenarios](#table-level-scenarios)
+    - [Entity Level Scenarios](#entity-level-scenarios)
+* [Additional Samples](#additional-samples)
+
+## Migration benefits
+
+As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Python client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) were created for all languages to drive a consistent experience with established API patterns for all services. A set of [Python guidelines](https://azure.github.io/azure-sdk/python_design.html) were also introduced to ensure that Python clients have a natural and idiomatic feel with respect to the Python ecosystem. The new `azure-data-tables` follows these guidelines.
+
+### Cross Service SDK improvements
+
```

Comment:
This section and the next section seem to describe actual benefits to a developer, i.e. new features, improved logging and authentication experience. Combine all these elements under a new heading titled 'Migration benefits" or "Why upgrade?"

Reply:
The formatting of these sections is taken from the template [here](https://github.com/Azure/azure-sdk/blob/main/docs/policies/migration-guide-template.md)

Reply:
I see. Who owns the template?

Reply:
I believe @ramya-rao-a

Reply:
I think the template could be tweaked but we have bigger fish to fry at the moment so I'll withdraw my comments for now. The doc LGTM